### PR TITLE
cmake-overhaul-v2-part-1

### PR DESCRIPTION
**Pull request by jrmadsen**
- Renames cmake files to be consistent, e.g. 
  - `cap_arch_def.cmake` -> `DyninstCapArchDef.cmake`
  - `Boost.cmake` -> `DyninstBoost.cmake`
- Places all scripts which serve find or build a third-party library in `cmake/tpls` folder
- Fixes inconsistency with `FindThreadDB.cmake` where the package is named `Thread_Db`

### CODE REVIEW

**Comment by jrmadsen** [comment]
So this PR is _definitely_ going to break the test suite but that's kind of the point. When all the changes are in, all the stuff like this:

```cmake
# Read the cache generated from building Dyninst
load_cache(${Dyninst_DIR}
           Boost_SYSTEM_LIBRARY_RELEASE
           Boost_INCLUDE_DIRS
           Boost_LIBRARY_DIRS
           Boost_DEFINES
           TBB_INCLUDE_DIRS
           TBB_LIBRARY_DIRS
           TBB_DEFINES
           ElfUtils_INCLUDE_DIRS
           ElfUtils_LIBRARY_DIRS
           STERILE_BUILD
           USE_GNU_DEMANGLER
           LibIberty_LIBRARY_DIRS
           )

# Import the include and library directory names
# NB: TBB and ElfUtils are not (currently) used directly in
#     Testsuite, but are transitively included from headers
#     in Dyninst.
include_directories(${Boost_INCLUDE_DIRS})
link_directories(${Boost_LIBRARY_DIRS})
add_definitions(${Boost_DEFINES})

include_directories(${TBB_INCLUDE_DIRS})
link_directories(${TBB_LIBRARY_DIRS})
add_definitions(${TBB_DEFINES})

include_directories(${ElfUtils_INCLUDE_DIRS})
link_directories(${ElfUtils_LIBRARY_DIRS})

# ... etc. ...

  target_link_libraries(testdyninst
                        testlaunch
                        testSuite
                        dyninstAPI
                        instructionAPI
                        common
                        ${Boost_SYSTEM_LIBRARY_RELEASE}
                        ${CMAKE_THREAD_LIBS_INIT})
```

Is entirely unnecessary. Everything w.r.t. the third-party libs (Boost/TBB/etc.) are provided through the target, e.g.: 

```cmake
target_link_libraries(testdyninst
    testlaunch
    testSuite
    Dyninst::dyninstAPI
    Dyninst::instructionAPI
    Dyninst::common
)
``` 

automatically appends the relevant flags for Boost, TBB, etc.

After all the changes I made in the PR that didn't get merged, this is what finding and using dyninst looked like:

```cmake
    find_package(Dyninst REQUIRED
                 COMPONENTS dyninstAPI parseAPI instructionAPI symtabAPI)

    add_executable(my_exe_using_dyninst PRIVATE Dyninst::Dyninst)
```

That. is. it.

**Comment by jrmadsen** [comment]
@hainest For reference, here is the project (which has finally been open-sourced) that we are using Dyninst for: [omnitrace](https://github.com/AMDResearch/omnitrace) and the [docs](https://amdresearch.github.io/omnitrace)

File: .github/workflows/libabigail.yaml [hunk 1]
```diff
@@ -31,12 +31,14 @@ jobs:
       uses: actions/checkout@v3
     - name: Build
       run: |
+         apt-get update
+         apt-get install -y python3 build-essential cmake
```
Comment (jrmadsen) [inline .github/workflows/libabigail.yaml:35]:
```suggestion
```

File: .github/workflows/libabigail.yaml [hunk 2]
```diff
@@ -31,12 +31,14 @@ jobs:
       uses: actions/checkout@v3
     - name: Build
       run: |
+         apt-get update
```
Comment (jrmadsen) [inline .github/workflows/libabigail.yaml:34]:
```suggestion
```

**Comment by hainest** [comment]
Subsumed by #1296

**Pull request closed without merging**